client/state/plugins/installed/status/reducer.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Internal dependencies
+ */
+import {
+	PLUGIN_ACTIVATE_REQUEST,
+	PLUGIN_ACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_ACTIVATE_REQUEST_FAILURE,
+	PLUGIN_DEACTIVATE_REQUEST,
+	PLUGIN_DEACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_DEACTIVATE_REQUEST_FAILURE,
+	PLUGIN_UPDATE_REQUEST,
+	PLUGIN_UPDATE_REQUEST_SUCCESS,
+	PLUGIN_UPDATE_REQUEST_FAILURE,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST_FAILURE,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST_FAILURE,
+	PLUGIN_INSTALL_REQUEST,
+	PLUGIN_INSTALL_REQUEST_SUCCESS,
+	PLUGIN_INSTALL_REQUEST_FAILURE,
+	PLUGIN_REMOVE_REQUEST,
+	PLUGIN_REMOVE_REQUEST_SUCCESS,
+	PLUGIN_REMOVE_REQUEST_FAILURE,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+/*
+ * Tracks the current status of plugins on sites, indexed by (site, plugin).
+ */
+export default function status( state = {}, action ) {
+	const { siteId } = action;
+	switch ( action.type ) {
+		case PLUGIN_ACTIVATE_REQUEST:
+		case PLUGIN_DEACTIVATE_REQUEST:
+		case PLUGIN_UPDATE_REQUEST:
+		case PLUGIN_AUTOUPDATE_ENABLE_REQUEST:
+		case PLUGIN_AUTOUPDATE_DISABLE_REQUEST:
+		case PLUGIN_INSTALL_REQUEST:
+		case PLUGIN_REMOVE_REQUEST:
+		case PLUGIN_ACTIVATE_REQUEST_SUCCESS:
+		case PLUGIN_DEACTIVATE_REQUEST_SUCCESS:
+		case PLUGIN_UPDATE_REQUEST_SUCCESS:
+		case PLUGIN_AUTOUPDATE_ENABLE_REQUEST_SUCCESS:
+		case PLUGIN_AUTOUPDATE_DISABLE_REQUEST_SUCCESS:
+		case PLUGIN_INSTALL_REQUEST_SUCCESS:
+		case PLUGIN_REMOVE_REQUEST_SUCCESS:
+		case PLUGIN_ACTIVATE_REQUEST_FAILURE:
+		case PLUGIN_DEACTIVATE_REQUEST_FAILURE:
+		case PLUGIN_UPDATE_REQUEST_FAILURE:
+		case PLUGIN_AUTOUPDATE_ENABLE_REQUEST_FAILURE:
+		case PLUGIN_AUTOUPDATE_DISABLE_REQUEST_FAILURE:
+		case PLUGIN_INSTALL_REQUEST_FAILURE:
+		case PLUGIN_REMOVE_REQUEST_FAILURE:
+			if ( state.hasOwnProperty( siteId ) ) {
+				return Object.assign( {}, state, {
+					[ siteId ]: statusForSite( state[ siteId ], action )
+				} );
+			}
+			return Object.assign( {}, state, { [ siteId ]: statusForSite( {}, action ) } );
```

Comment:
Couldn't this case block be simplified to:

```js
Object.assign( {}, state, { [ siteId ]: statusForSite( state[ siteId ], action ) } );
```

Since even if `state.hasOwnProperty( siteId )` is `false`, the `statusForSite` default argument for `state` will provide the empty object.